### PR TITLE
feat(RunContext): toPromise resolve targetDirectory instead of nothing

### PR DESCRIPTION
**Pull request by Awk34**

### CODE REVIEW

File: lib/run-context.js [hunk 1]
```diff
@@ -119,7 +119,9 @@ RunContext.prototype._run = function () {
  */
 RunContext.prototype.toPromise = function () {
   return new Promise(function (resolve, reject) {
-    this.on('end', resolve);
+    this.on('end', function () {
+      resolve(this.targetDirectory);
```
Comment (SBoudrias) [inline lib/run-context.js]: `this` is undefined in this case, you'll need to bind a context to this function.

Comment (Awk34) [inline lib/run-context.js]: Actually, the `listener` functions of Node `EventEmitter`s are bound to the same EventEmitter, so this works here

Comment (Awk34) [inline lib/run-context.js]: But if you think that it makes more sense to bind it to `this`, I would understand

Comment (SBoudrias) [inline lib/run-context.js]: I think it would be less confusing when reading the code.

Comment (Awk34) [inline lib/run-context.js]: Updated. I was getting that error on plain 1.3.0. The updated code fails on all 4 versions. I'm not sure what's wrong there, but I'll look into it.

**Comment by SBoudrias** [comment]
I think the tests failing are due to the bump in yeoman-generator/yeoman-environment having a side effect here.

**Pull request merged**